src/mesh/bc_mesh.cc

```diff
@@ -86,7 +81,58 @@ unsigned int BCMesh::n_nodes() const {
 
 
 ElementAccessor<3> BCMesh::element_accessor(unsigned int idx) const {
```

Comment:
Rather duplicate some data structures in BCMesh instance and move these intensively used getters and accessors to the MeshBase.